### PR TITLE
Calendar aria updates

### PR DESCRIPTION
**Pull request by Fercas123**
- Calendar aria-label format changed from 'MM YYYY' to 'Calendar, DD MMM YYYY'.
- Calendar day aria-label format changed from 'DD/MM/YYYY' to 'DD MMM YYYY'.
- Calendar navigation buttons aria-label format got simplified to 'Previous Month' and 'Next Month'.

### CODE REVIEW

File: packages/lab/src/calendar/Calendar.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const Calendar = forwardRef<HTMLDivElement, CalendarProps>(
         <div
           className={clsx(withBaseName(), className)}
           role="application"
-          aria-label={calendarLabel}
+          aria-label={`Calendar, ${calendarLabel}`}
```
Comment (joshwooding) [inline packages/lab/src/calendar/Calendar.tsx:76]: Can we move "Calendar" to the modal overlay

File: packages/lab/src/calendar/internal/CalendarNavigation.tsx [hunk 2]
```diff
@@ -283,11 +279,7 @@ export const CalendarNavigation = forwardRef<
           onClick={handleNavigateNext}
           focusableWhenDisabled={true}
         >
-          <ChevronRightIcon
-            aria-label={`Next Month, ${formatDate(
-              visibleMonth.add({ months: 1 })
-            )}`}
-          />
+          <ChevronRightIcon aria-label={"Next Month"} />
```
Comment (joshwooding) [inline packages/lab/src/calendar/internal/CalendarNavigation.tsx:282]:
```suggestion
          <ChevronRightIcon aria-label="Next Month" />
```

File: packages/lab/src/calendar/Calendar.tsx (hunk 1 above)
Comment (Fercas123) [reply packages/lab/src/calendar/Calendar.tsx:76]: you mean out of calendar and in date picker instead?

Comment (joshwooding) [reply packages/lab/src/calendar/Calendar.tsx:76]: Yes please

File: packages/lab/src/__tests__/__e2e__/calendar/Calendar.cy.tsx [hunk 3]
```diff
@@ -33,22 +33,30 @@ const formatDate = (date: DateValue, options?: Intl.DateTimeFormatOptions) => {
...
   return formatter.format(date.toDate(localTimeZone));
 };
+const dayOptions: Intl.DateTimeFormatOptions = {
+  day: "2-digit",
+  month: "long",
+  year: "numeric",
+};
+
+const formatDay = (date: DateValue) => {
+  return formatDate(date, dayOptions);
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/calendar/Calendar.cy.tsx:43]: Is there any benefit to having dayOptions in a variable?

File: packages/lab/src/calendar/useCalendar.ts [hunk 4]
```diff
@@ -143,7 +143,7 @@ export function useCalendar(props: useCalendarProps) {
   const getInitialFocusedDate = (): DateValue => {
     const selectedDate = selectionManager.state.selectedDate;
     // Case range or offset
-    if (isRangeOrOffsetSelectionValue(selectedDate)) {
+    if (isRangeOrOffsetSelectionWithValues(selectedDate)) {
```
Comment (joshwooding) [inline packages/lab/src/calendar/useCalendar.ts:146]: Do we need to check if the endDate is present for this logic? What happens if you start a range selection then press tab?

Comment (Fercas123) [reply packages/lab/src/calendar/useCalendar.ts:146]: 💥

File: packages/lab/src/calendar/useSelection.ts [hunk 5]
```diff
@@ -90,6 +90,15 @@ export function isRangeOrOffsetSelectionValue(
   return selectionValue != null && isPlainObject(selectionValue);
 }
 
+export function isRangeOrOffsetSelectionWithValues(
+  selectionValue?: AllSelectionValueType
+): selectionValue is RangeSelectionValueType | OffsetSelectionValueType {
+  return (
+    isRangeOrOffsetSelectionValue(selectionValue) &&
+    Boolean(selectionValue?.startDate) &&
```
Comment (Fercas123) [inline packages/lab/src/calendar/useSelection.ts:98]:
```suggestion
    Boolean(selectionValue?.startDate)
```

File: packages/lab/src/calendar/useSelection.ts [hunk 6]
```diff
@@ -90,6 +90,15 @@ export function isRangeOrOffsetSelectionValue(
   return selectionValue != null && isPlainObject(selectionValue);
 }
 
+export function isRangeOrOffsetSelectionWithValues(
```
Comment (joshwooding) [inline packages/lab/src/calendar/useSelection.ts:93]:
```suggestion
export function isRangeOrOffsetSelectionWithStartDate(
```
?

**Review by origami-z — APPROVED** [review]
Aria label reminded me that users asked before to provide customized label (mostly for i18n purposes) for all internal labels.

**Review by lilyvc — APPROVED** [review]

**Pull request merged**